protocol/contracts/beanstalk/sun/SeasonFacet/Distribution.sol

```diff
@@ -0,0 +1,164 @@
+            //     )
+            // );
+            receiveShipment(
+                shipmentRoutes[i].recipient,
+                shipmentAmounts[i],
+                shipmentRoutes[i].data
+            );
+        }
+
+        emit Ship(s.season.current, shipmentRoutes, shipmentAmounts);
+    }
+
+    /**
+     * @notice Determines the amount of Beans to distribute to each shipping route based on points.
+     * @dev Does not factor in route cap.
+     * @dev If points are 0, does not alter the associated shippingAmount.
+     * @dev Assumes shipmentAmounts and shipmentRoutes have matching shape and ordering.
+     */
+    function getBeansFromPoints(
+        uint256[] memory shipmentAmounts,
+        ShipmentPlan[] memory shipmentPlans,
+        uint256 totalPoints,
+        uint256 beansToShip
+    ) private pure {
+        for (uint256 i; i < shipmentPlans.length; i++) {
+            // Do not modify amount for streams with 0 points. They either are zero or have already been set.
+            if (shipmentPlans[i].points == 0) continue;
+            shipmentAmounts[i] = (beansToShip * shipmentPlans[i].points) / totalPoints; // round down
+        }
+    }
+
+    function hashShipmentRoute(
+        Storage.ShipmentRoute memory shipmentRoute
+    ) public pure returns (bytes32 hash) {
+        return
+            keccak256(
+                abi.encode(
+                    shipmentRoute.planContract,
+                    shipmentRoute.planSelector,
+                    uint8(shipmentRoute.recipient),
+                    keccak256(shipmentRoute.data)
+                )
+            );
+    }
+
+    /**
+     * @notice Gets the shipping plan for all shipping routes.
+     * @dev Determines which routes are active and how many Beans they will receive.
+     * @dev GetPlan functions should never fail/revert. Else they will have no Beans allocated.
+     */
+    function getShipmentPlans(
+        Storage.ShipmentRoute[] memory shipmentRoutes
+    ) private view returns (ShipmentPlan[] memory shipmentPlans, uint256 totalPoints) {
+        shipmentPlans = new ShipmentPlan[](shipmentRoutes.length);
+        for (uint i; i < shipmentRoutes.length; i++) {
+            (bool success, bytes memory returnData) = shipmentRoutes[i].planContract.staticcall(
+                abi.encodeWithSelector(shipmentRoutes[i].planSelector, shipmentRoutes[i].data)
+            );
+            if (success) {
+                shipmentPlans[i] = abi.decode(returnData, (ShipmentPlan));
+            } else {
+                shipmentPlans[i] = ShipmentPlan({points: 0, cap: 0});
+            }
+            totalPoints += shipmentPlans[i].points;
+        }
+    }
+
+    /**
+     * @notice Replaces the entire set of ShipmentRoutes with a new set.
+     * @dev Changes take effect immediately and will be seen at the next sunrise mint.
+     */
+    function setShipmentRoutes(Storage.ShipmentRoute[] calldata shipmentRoutes) external {
```

Comment:
Could we also have view functions for the shipment route in storage? Defer to you on whether we want to also include functions that expose the individual values as well as the struct.

Reply:
added here
https://github.com/BeanstalkFarms/Beanstalk/pull/860/commits/212e19e040586cc4b4b2e4404b85099613344ab3#diff-a3ff8fea8dfdc60875a144d7d0f1fb3b1e5f0b82f2cbcb1331f2fca38d73f833R153-R159